src/ui/components/AccountSelector.js

```diff
@@ -29,22 +27,45 @@ export default function AccountDropdown({
   ledger,
 }: DropdownProps) {
   const items = ledger.accounts().filter((a) => a.active);
+  const [selectedAccount, setSelectedAccount] = useState<Account | null>(null);
 
-  const onComboChange = (e, inputObj) => setCurrentIdentity(inputObj);
+  const onComboChange = (e, inputObj) => {
+    setSelectedAccount(inputObj);
+    setCurrentIdentity(inputObj);
+  };
+
+  const renderInput = (params) => {
+    const name = selectedAccount ? selectedAccount.identity.name : "";
+    const balStr = selectedAccount
+      ? ` (${format(selectedAccount.balance, 2)})`
+      : "";
+    return (
+      <TextField
+        fullWidth
+        {...params}
+        inputProps={{...params.inputProps, value: `${name}${balStr}`}}
+        label={placeholder}
+      />
+    );
+  };
 
   return (
     <ThemeProvider theme={theme}>
       <Autocomplete
         onChange={onComboChange}
         fullWidth
         options={items}
-        getOptionLabel={(item) =>
-          `${item.identity.name} (${format(item.balance, 2)})`
-        }
-        style={{margin: "20px"}}
-        renderInput={(params) => (
-          <TextField fullWidth {...params} label={placeholder} />
-        )}
+        getOptionLabel={(item: Account) => {
+          const balStr = format(
+            selectedAccount && selectedAccount.balance !== fromInteger(0)
+              ? selectedAccount.balance
+              : item.balance,
```

Comment:
Can you help me understand what's the distinction between the selectedAccount and the item, and why we would want to render the balance for the selectedAccount rather than the item?

Reply:
This is primarily to have the balances on the menu items within the combobox prior to selecting an item. The way material-ui works the labels inside the dropdown menu are the same as the values displayed on select, so this was the most sensible way to do this I think. The item though does not update when the the balances change, so we need to use the selectedAccount for that.

Reply:
So there were some weird sync issues I was seeing after transferring grain, and having account balances adjust. Removing balance logiv from the Autocomplete components fixed everything and cleaned up the code a lot, getting rid of these ugly code patterns you saw in the process.